### PR TITLE
Fix/smart records merge on create

### PR DESCRIPTION
**Pull request by ifree92**

### CODE REVIEW

File: docs/resources/domain_records.md [hunk 1]
```diff
@@ -60,3 +60,5 @@ resource "namecheap_domain_records" "my-domain2-com" {
 - `type` - (Required) Possible values: A, AAAA, ALIAS, CAA, CNAME, MX, MXE, NS, TXT, URL, URL301, FRAME
 - `mx_pref` - (Optional) MX preference for host. Applicable for MX records only
 - `ttl` - (Optional) Time to live for all record types. Possible values: any value between 60 to 60000
+
+~> It is strongly recommended to set `address` and `hostname` in lower case to prevent undefined behavior!
```
Comment (StyleT) [inline docs/resources/domain_records.md:64]: Can we always convert them to lowercase automatically during comparison?

File: docs/guides/namecheap_domain_records_guide.md [hunk 2]
```diff
@@ -34,6 +34,9 @@ The same workflow works for both: `record` items and `nameservers`.
 
 -> This is the default behavior, however, we recommend to set the mode explicitly.
 
+~> Upon creating a new resource with the records or nameservers that already exist, the provider will throw a duplicate
```
Comment (StyleT) [inline docs/guides/namecheap_domain_records_guide.md:37]: I think we need to add that it happens if subdomain & record type match

**Review by StyleT — CHANGES_REQUESTED** [review]
Let's cover existing code with some basic tests & merge this PR with tests that are covering the issue.

File: docs/guides/namecheap_domain_records_guide.md (hunk 2 above)
Comment (ifree92) [reply docs/guides/namecheap_domain_records_guide.md:37]: Noted. I will check this.
At the moment, we have duplicates comparison only by `hostname` (subdomain), `type`, and `address`.

However, user allowed to create "mistakes" like:

```terraform
resource "namecheap_domain_records" "domain" {
   #...
   record {
     hostname = "dev"
     type = "A"
     address = "11.22.33.44"
   }

   record {
     hostname = "dev"
     type = "A"
     address = "33.33.33.33"
   }
}
```

I will amend this behavior (and provide tests for all cases! 😆 )

File: docs/resources/domain_records.md (hunk 1 above)
Comment (ifree92) [reply docs/resources/domain_records.md:64]: Basically, yes.
However, the existing behavior works in the way where if a user changes the camel case of the address of any record, the provider will "re-create" the record. I will have to make the READ smarter.

I will revert on this in the context of a current PR.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -24,6 +25,100 @@ jobs:
...
+          go version
+          go env GOPATH
+
+      - name: Acceptance Test
+        run: |
+          export CGO_ENABLED=0
+          source go-env.sh
+          make testacc
+        env:
+          NAMECHEAP_USER_NAME: ${{ secrets.NAMECHEAP_USER_NAME }}
```
Comment (StyleT) [inline .github/workflows/ci.yml:98]: Let's move "non-secret" var values to this file. It will simplify debugging

**Review by StyleT — APPROVED** [review]

**Pull request merged**